tests/test_run.py

```diff
@@ -316,3 +317,35 @@ def test_local_args():
     print(state)
     print(log)
     assert log.find(", '--xyz', '789']") != -1, "params not detected in argv"
+
+
+def test_local_context():
+    global_context.set(None)
```

Comment:
That should probaly sit in `config_test_base` together with the rest of the cleanups